### PR TITLE
Remove code in __init__ that's causing CI failures downstream

### PR DESCRIPTION
**Pull request by rosteen**
This code in `__init__` is causing deprecation warnings (and thus CI failures) downstream when installing `drizzlepac` in CI. See for example https://github.com/spacetelescope/slitlessutils/actions/runs/9083277319/job/24961746510?pr=72. @pllim removed the same code from `stsci.tools` a while ago, and I'll be opening parallel PRs in a couple other packages.

### CODE REVIEW

File: stsci/__init__.py [hunk 1]
```diff
@@ -1,11 +0,0 @@
...
-    # As long as we're using setuptools/distribute, we need to do this the
-    # setuptools way or else pkg_resources will throw up unncessary and
-    # annoying warnings (even though the namespace mechanism will still
-    # otherwise work without it).  Get rid of this as soon as
-    # setuptools/distribute is dead.
-    __import__('pkg_resources').declare_namespace(__name__)
-except ImportError:
-    pass
-
-__path__ = __import__('pkgutil').extend_path(__path__, __name__)
```
Comment (pllim) [inline stsci/__init__.py:11]: What is this `__path__` used for?

While we're at this, should we add a `__all__` and include `__version__` like other packages, @zacharyburnett ?

Comment (pllim) [reply stsci/__init__.py:11]: Oh wait, this is namespace package level... `stsci.tools` has an extra `lib` folder. My head hurts.

Comment (rosteen) [reply stsci/__init__.py:11]: Yup 😅

Comment (rosteen) [reply stsci/__init__.py:11]: > What is this `__path__` used for?

I'm not sure but you removed it in stsci.tools in https://github.com/spacetelescope/stsci.tools/pull/144.

Comment (pllim) [reply stsci/__init__.py:11]: I just dunno if the presence of `lib/` folder matters or not. I pinged the SCSB packaging experts to review. 🤞 

Must have been painful to smoke this out. Have a virtual beer. 🍺

Comment (rosteen) [reply stsci/__init__.py:11]: Yeah it wasn't great, it looked like the failure was from `stsci.tools` but you'd already removed this code from there so I couldn't figure out where it was happening. I'm also making PRs to `stsci.imagestats` and `stsci.stimage` in addition to the two I made already.

Comment (zacharyburnett) [reply stsci/__init__.py:11]: oof, this namespace stuff with `stsci.` makes my head hurt as well. As far as I can tell this `__path__` is not needed... ?

As far as `__version__`, that should be in the metadata (available via `importlib.metadata`), but if we do ever need to reference the version from within the source, `setuptools_scm` can automatically generate and install a version file for you if you add the configuration to its section in `pyproject.toml`: https://github.com/spacetelescope/stsci.image/blob/e04b39b229a835fcaa596e00f9b900d96f057a98/pyproject.toml#L56
like so:
```toml
[tool.setuptools_scm]
version_file = "stsci/image/version.py"
```
then (after a `pip install`) you can reference it like:
```python
from .version import __version__
```
or like
```python
from stsci.image.version import __version__
```

**Review by zacharyburnett — APPROVED** [review]

**Comment by pllim** [comment]
Probably needs a release with this patch.

**Pull request merged**